replica/database.cc

```diff
@@ -1940,7 +1940,11 @@ future<> database::do_apply(schema_ptr s, const frozen_mutation& m, tracing::tra
 
     row_locker::lock_holder lock;
     if (!cf.views().empty()) {
-        auto lock_f = co_await coroutine::as_future(cf.push_view_replica_updates(s, m, timeout, std::move(tr_state), get_reader_concurrency_semaphore()));
+        if (!_view_update_generator) {
+            co_await coroutine::return_exception(std::runtime_error("view update generator not plugged to push updates"));
```

Comment:
I think you should just log an rate-limited warning/error here, unless synchronous update are on. We made a conscious design decision on not making regular writes depend on the view updates they generate by default. Failing them because an update cannot be generated would violate this principle.

Reply:
@nyh I'd like a second opinion on this.

Reply:
> We made a conscious design decision on not making regular writes depend on the view updates they generate by default. 

But if `cf.push_view_replica_updates` returns exceptional future the whole `databasE::do_apply` aborts and returns back the very same exception. What am I missing?

Reply:
Maybe it doesn't -- it swallows exceptions? I'm just hand waving here. We need someone more familiar with this code like @nyh to chime in.

Reply:
I think this exception is justified. If I understand correctly, if this exception ever happens, it's some sort of a bug - it could almost be a on_internal_error(), because we always expect this vug to have been set up long before the view update is generated.
Moreover, if there is some sort of race condition (e.g., a couple of writes sneak in just before we set this up), it's better for the whole write to be failed on "internal server error" (and, presumably, retried by the application) than to have the base write done and the view update just ignored.
So I think this exception is the right thing to do.

Reply:
Ok, fair enough.